Classes/Service/WallsService.php

```diff
@@ -270,14 +256,15 @@ protected function getSanitizedPost(array $post, PluginConfiguration $pluginConf
     protected function cacheExternalResources(string $resource, int $contentRecordUid): string
     {
         $pathParts = GeneralUtility::split_fileref(parse_url($resource, PHP_URL_PATH));
+
         $filePath = sprintf(
             '%s%s.%s',
             $this->getTargetDirectory($contentRecordUid),
             $pathParts['filebody'],
             $pathParts['fileext']
         );
 
-        if (!file_exists($filePath)) {
+        if (!file_exists($filePath) && GeneralUtility::getUrl($resource)) {
             GeneralUtility::writeFile($filePath, GeneralUtility::getUrl($resource));
         }
```

Comment:
Your code requested external API twice:

```suggestion
        if (
            !file_exists($filePath)
            && $responseContent = GeneralUtility::getUrl($resource)
        ) {
            GeneralUtility::writeFile($filePath, $responseContent);
        }
```